main.py

```diff
@@ -64,6 +64,7 @@ def __init__(self):
     self.result = False
     self.metadata = None
     self.fatal_error = None
+    self.data_filepath = None
```

Comment:
Can we document these attributes? Glancing at it, I'm realizing I can't figure out what any of them mean and can't infer the likely data types.